### PR TITLE
search panel added

### PR DESCRIPTION
**Pull request by Manikant25**
**Fixes**: 
solves #385 

**Description:**
Created search pannel in the patient list view in the reference app. Further functionality will be added on approval from the reviewer.

**Type**
Feature

**Screenshots (if applicable)**
rsz_screenshot_20210324-105703

rsz_1screenshot_20210324-110006

### CODE REVIEW

**Comment by deepankarb** [comment]
Hi @Manikant25, thanks for the PR. I suggest moving it to "draft" state until it is ready for review.

File: reference/src/main/java/com/google/android/fhir/reference/PatientListActivity.kt [hunk 1]
```diff
@@ -35,7 +35,7 @@ import com.google.android.fhir.reference.FhirApplication.Companion.fhirEngine
 import com.google.android.material.snackbar.Snackbar
 
 /** An activity representing a list of Patients. */
-class PatientListActivity : AppCompatActivity(), SearchView.OnQueryTextListener {
+class PatientListActivity : AppCompatActivity(),SearchView.OnQueryTextListener {
```
Comment (epicadk) [inline reference/src/main/java/com/google/android/fhir/reference/PatientListActivity.kt:38]: This change isn't needed. This is probably why the spotless check is failing.

**Comment by Manikant25** [comment]
> Hi @Manikant25, thanks for the PR. I suggest moving it to "draft" state until it is ready for review.

okay

**Comment by jingtang10** [comment]
Thanks for showing the mocks. Please note that the design isn't finalized and this isn't ready to be worked on. I think what we'll likely have is some kind of side/bottom panel for searching by specific fields.

**Comment by Manikant25** [comment]
> Thanks for showing the mocks. Please note that the design isn't finalized and this isn't ready to be worked on. I think what we'll likely have is some kind of side/bottom panel for searching by specific fields.

That's a good idea, so may I close this PR now.

**Comment by jingtang10** [comment]
Can you add a side panel like this: https://material.io/components/sheets-side#usage?

In the side panel please add a first name field and a last name field.

https://github.com/OpenSRP/fhircore/issues/50

**Comment by Manikant25** [comment]
@jingtang10, Yes I can add.

**Comment by jingtang10** [comment]
Hi @Manikant25 do you plan to update this PR or should I close? Either way, thanks for your contribution!

**Comment by Manikant25** [comment]
@jingtang10 Sorry for the delay I was caught up with my exams.  Actually, I did not find its implementation or any sample app so it took time. I will update this PR soon.

**Comment by Manikant25** [comment]
I could not impelemnt [Side:sheet](https://material.io/components/sheets-side#usage) as its implementation is not given nor there is a sample app using it. However, I tried to make using the menu. Let me know if it's fine. @jingtang10 

**This is the sample** 
screenshot_20210410-022020

**Comment by jingtang10** [comment]
Thanks @Manikant25.

We're clearly not going to be search first/last names using a checkbox. Can you get some inspiration from this app: https://github.com/material-components/material-components-android? the side navigation panel might work? or of not use the bottom drawer.

**Comment by Manikant25** [comment]
Yes, I look into it.

**Comment by Manikant25** [comment]
[ Bottom-sheet ](https://material.io/components/sheets-bottom#anatomy) looks good to me. If we can add a switch for first name and lastname. If you allow may I start. @jingtang10

**Comment by epicadk** [comment]
@Manikant25 are you working on this?

**Comment by Manikant25** [comment]
@epicadk Yes, I am working on it.

**Comment by jingtang10** [comment]
@Manikant25 closing this since @aditya-07 is working on this in #518. Thanks!

**Pull request closed without merging**